julearn/transformers/tests/test_confounds.py

```diff
@@ -2,65 +2,149 @@
 #          Sami Hamdan <s.hamdan@fz-juelich.de>
 # License: AGPL
 import pandas as pd
+from pandas.testing import assert_frame_equal
 import numpy as np
+from numpy.testing import assert_array_equal
 
 from julearn.transformers.confounds import DataFrameConfoundRemover
-
-
-X = pd.DataFrame({'A': np.arange(10), 'B': np.arange(10, 20),
-                  'cheese__:type:__confound': np.arange(50, 60)})
-
-X_multi = pd.DataFrame({'A': np.arange(10), 'B': np.arange(10, 20),
-                        'cookie__:type:__confound': np.arange(40, 50),
-                        'cheese__:type:__confound': np.arange(50, 60)})
-
-
-def test_DataFrameConfoundRemover_suffix_one_conf_no_error():
-
-    remover = DataFrameConfoundRemover()
-    X_trans = remover.fit_transform(X)
-    actual = X_trans.columns
-    expected = X[['A', 'B']].columns
-    assert len(actual) == len(expected)
-    assert all([a == b for a, b in zip(actual, expected)])
-
-
-def test_DataFrameConfoundRemover_suffix_multi_conf_no_error():
-
-    remover = DataFrameConfoundRemover()
-    X_trans = remover.fit_transform(X_multi)
-    actual = X_trans.columns
-    expected = X[['A', 'B']].columns
-    assert len(actual) == len(expected)
-    assert all([a == b for a, b in zip(actual, expected)])
-
-
-def test_DataFrameConfoundRemover_no_suffix_one_conf_str_no_error():
-
-    remover = DataFrameConfoundRemover(confounds='cheese__:type:__confound')
-    X_trans = remover.fit_transform(X)
-    actual = X_trans.columns
-    expected = X[['A', 'B']].columns
-    assert len(actual) == len(expected)
-    assert all([a == b for a, b in zip(actual, expected)])
-
-
-def test_DataFrameConfoundRemover_no_suffix_one_conf_list_no_error():
-
-    remover = DataFrameConfoundRemover(confounds=['cheese__:type:__confound'])
-    X_trans = remover.fit_transform(X)
-    actual = X_trans.columns
-    expected = X[['A', 'B']].columns
-    assert len(actual) == len(expected)
-    assert all([a == b for a, b in zip(actual, expected)])
-
-
-def test_DataFrameConfoundRemover_no_suffix_mutli_conf_list_no_error():
-
-    remover = DataFrameConfoundRemover(
-        confounds=['B', 'cheese__:type:__confound'])
-    X_trans = remover.fit_transform(X)
-    actual = X_trans.columns
-    expected = X[['A']].columns
-    assert len(actual) == len(expected)
-    assert all([a == b for a, b in zip(actual, expected)])
+from sklearn.linear_model import LinearRegression
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.base import clone
+
+X = pd.DataFrame({
```

Comment:
will this be the standard way to define columns for confound removal to work?

Reply:
This is what the pipeline is doing internally. 
Change the column names (only internally) . 
As you can see in the tests you can just provide a list of any column_names, which is then used as the confounds. 

So internally yes using suffixes etc. is the general way.
But if you only want to use confound removal without the rest, you could just input the column_names instead. I would consider this standard for end users and also give an example how to do it.

Reply:
Got it!